src/app.rs

```diff
@@ -10,22 +10,44 @@ use {
         terminal, ExecutableCommand, QueueableCommand,
     },
     notify::{RecommendedWatcher, RecursiveMode, Watcher},
-    std::{
-        env,
-        fs,
-        io::Write,
-        path::PathBuf,
-    },
+    std::{env, fs, io::Write, path::PathBuf, process::Command},
     termimad::{Event, EventSource},
+    cargo_metadata::{Metadata, MetadataCommand}
 };
 
+
+fn find_folders_to_watch(cargo_filepath: &PathBuf) -> Result<Vec<PathBuf>> {
+    let metadata = MetadataCommand::new().manifest_path(cargo_filepath).exec()?;
+
+    let mut folders_to_watch = vec![];
+
+    for item in metadata.packages {
+        if item.source.is_none() {
+            // We're only concerned with items where the source is None
+            for target in item.targets {
+                if target.src_path.ends_with("lib.rs") || target.src_path.ends_with("main.rs") {
+                    // targets can contain the build script as well, so we need to eliminate them.
+
+                    let target_folder = target.src_path.parent().expect("parent of a target folder is a root folder");
+                    if target_folder.ends_with("src") {
```

Comment:
This also doesn't seem right, `src` is just a convention and not required.

Reply:
This is just to follow the convention defined by `bacon`, which is to check `src/` directory. The idea I had, I guess, of this MR was to _extend_ the usage to workspaces, but not necessarily change the `semantics` of what's being watched.

Open to suggestions, though, on what exactly to watch, which we could address in a separate MR.